client/src/pages/register/register.jsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from 'react';
+import classes from './register.module.css';
+import { RegisterForm } from './RegisterForm';
+import { Flex } from '@mantine/core';
+
+/**
+ * Register page component
+ */
+function Register() {
+  const [user, setUser] = useState({
+    firstName: '',
+    middleName: '',
+    lastName: '',
+    email: '',
+    password: '',
+    licenseNumber: '',
+  });
+  const [errors, setErrors] = useState({
+    firstName: '',
+    middleName: '',
+    lastName: '',
+    email: '',
+    password: '',
+    licenseNumber: '',
+  });
+
+  const [showLicenseHelper, setShowLicenseHelper] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [formState, setFormState] = useState(1);
+
+  /**
+   * Handles input fields in the Registration form
+   * @param {*} event text field events
+   */
+  function onFormChange(event) {
+    const newUserChange = {},
+      newErrorsChange = {};
+    newUserChange[event.target.name] = event.target.value;
+    newErrorsChange[event.target.name] = '';
+    setUser({ ...user, ...newUserChange });
+    setErrors({ ...errors, ...newErrorsChange });
+  }
+
+  /**
+   * Checks if License Number is a valid license number.
+   * Toggles Error if invalid.
+   */
+  function fetchLicenseNumber() {
+    setIsLoading(true);
+    fetch('/api/v1/licenses?license=' + user.licenseNumber)
+      .then((response) => {
+        if (!response.ok) {
+          return Promise.reject(response);
+        }
+        return response.json();
+      })
+      .then((license) => {
+        if (license) {
+          setUser({ ...user, licenseNumber: license.licenseNumber });
+          setFormState(2);
+          setShowLicenseHelper(false);
+        }
+      })
+      .catch((error) => {
+        setShowLicenseHelper(true);
+        error.json().then((body) => {
+          setErrors({ ...errors, licenseNumber: body.message });
+        });
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }
+
+  /**
+   * Process the Registration with the Register API
+   * Toggles any field errors if any is invalid
+   */
+  function handleRegister() {
+    fetch('/api/v1/users/register', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(user),
+    })
+      .then((response) => {
+        if (!response.ok) {
+          return Promise.reject(response);
+        }
+        return response.json();
+      })
+      .then((result) => {
+        console.log('success', result);
```

Comment:
Can we add a confirmation message, like we see in the prototype design?

https://www.figma.com/design/C2kceQP9lzTPMPnywXZd9t/Public-playground--SF-Life-Line?node-id=1481-6634&t=zSYgWyzy4F26bjyF-4